File: backend/app/schemas/calculator.py
from pydantic import BaseModel, Field
from typing import List, Optional
from enum import Enum
from datetime import date

class PropertyType(str, Enum):
    RESIDENTIAL = "residential"
    COMMERCIAL = "commercial"
    INDUSTRIAL = "industrial"

class AssetCategory(str, Enum):
    CAPITAL_WORKS = "capital_works"
    PLANT_EQUIPMENT = "plant_equipment"

class AssetItem(BaseModel):
    name: str
    category: AssetCategory
    value: float
    effective_life: Optional[int] = None

class DepreciationRequest(BaseModel):
    property_type: PropertyType
    construction_date: date
    purchase_date: date
    purchase_price: float
    is_new_property: bool
    assets: List[AssetItem] = Field(default_factory=list)

class DepreciationYearDetail(BaseModel):
    year: int
    diminishing_value: float
    prime_cost: float
    capital_works: float
    total: float

class DepreciationResponse(BaseModel):
    property_type: PropertyType
    purchase_price: float
    total_depreciable_amount: float
    yearly_breakdown: List[DepreciationYearDetail]
    first_year_depreciation: float
    five_year_depreciation: float 

File: frontend/src/types/calculator.ts
export enum PropertyType {
  RESIDENTIAL = "residential",
  COMMERCIAL = "commercial",
  INDUSTRIAL = "industrial"
}

export enum AssetCategory {
  CAPITAL_WORKS = "capital_works",
  PLANT_EQUIPMENT = "plant_equipment"
}

export interface AssetItem {
  name: string;
  category: AssetCategory;
  value: number;
  effective_life?: number;
}

export interface DepreciationRequest {
  property_type: PropertyType;
  construction_date: Date;
  purchase_date: Date;
  purchase_price: number;
  is_new_property: boolean;
  assets: AssetItem[];
}

export interface DepreciationYearDetail {
  year: number;
  diminishing_value: number;
  prime_cost: number;
  capital_works: number;
  total: number;
}

export interface DepreciationResponse {
  property_type: PropertyType;
  purchase_price: number;
  total_depreciable_amount: number;
  yearly_breakdown: DepreciationYearDetail[];
  first_year_depreciation: number;
  five_year_depreciation: number;
} 